Intermittent failure in `test___non_streaming_measurement_execution___cancel___cancels_measurement`

# Bug Report

This test failed during a dependency update: https://github.com/ni/measurement-plugin-python/pull/1171
```
================================== FAILURES ===================================
__ test___non_streaming_measurement_execution___cancel___cancels_measurement __

measurement_plugin_client_module = <module 'test_measurement_client.py' from 'C:\\Users\\ec2-user\\AppData\\Local\\Temp\\pytest-of-unknown\\pytest-8\\measurement_plugin_client_files9\\test_measurement_client.py'>

    def test___non_streaming_measurement_execution___cancel___cancels_measurement(
        measurement_plugin_client_module: ModuleType,
    ) -> None:
        test_measurement_client_type = getattr(measurement_plugin_client_module, "TestMeasurement")
        measurement_plugin_client = test_measurement_client_type()
    
        with pytest.raises(grpc.RpcError) as exc_info:
            with concurrent.futures.ThreadPoolExecutor() as executor:
                measure = executor.submit(measurement_plugin_client.measure)
                measurement_plugin_client.cancel()
>               measure.result()
E               Failed: DID NOT RAISE <class 'grpc.RpcError'>

C:\ar\_work\measurement-plugin-python\measurement-plugin-python\packages\generator\tests\acceptance\test_streaming_measurement_client.py:81: Failed
- generated xml file: C:\ar\_work\measurement-plugin-python\measurement-plugin-python\packages\generator\test_results\nimg-i-00fce2112249175a2-py39.xml -
=========================== short test summary info ===========================
FAILED tests/acceptance/test_streaming_measurement_client.py::test___non_streaming_measurement_execution___cancel___cancels_measurement
=================== 1 failed, 30 passed in 68.68s (0:01:08) ===================
py39: exit 1 (75.55 seconds) C:\ar\_work\measurement-plugin-python\measurement-plugin-python\packages\generator> poetry run pytest -v --cov=ni_measurement_plugin_sdk_generator --cov-append --cov-report= --junitxml=test_results/nimg-i-00fce2112249175a2-py39.xml pid=8932
py39: FAIL ✖ in 1 minute 40.45 seconds
```

## Repro or Code Sample

https://github.com/ni/measurement-plugin-python/pull/1171

## Expected Behavior

It passes

## Current Behavior

It fails for some runs but not others.

## Possible Solution

<!--- Not obligatory, but suggest a fix/reason for the bug -->
<!--- Please let us know if you'd be willing to contribute the fix; we'd be happy to work with you -->

## Context

Test failure in unrelated PR.

## Your Environment

* OS & Device: Windows
* `ni-measurement-plugin-sdk-service` version: current main
* InstrumentStudio version: unknown
* Python version: 3.9


[AB#3162086](https://dev.azure.com/ni/94b22d7b-ad7b-4f5e-88f0-867910f91c94/_workitems/edit/3162086)

When it fails, it takes 10 seconds, which is the expected duration of the streaming test measurement.

`--log-level debug` shows:

```
tests/acceptance/test_streaming_measurement_client.py::test___non_streaming_measurement_execution___cancel___cancels_measurement[79-10000] FAILED [  0%]

====================================================== FAILURES =======================================================
_________________ test___non_streaming_measurement_execution___cancel___cancels_measurement[79-10000] _________________

measurement_plugin_client_module = <module 'test_measurement_client.py' from 'C:\\Users\\Administrator\\AppData\\Local\\Temp\\pytest-of-Administrator\\pytest-9\\measurement_plugin_client_files0\\test_measurement_client.py'>

    def test___non_streaming_measurement_execution___cancel___cancels_measurement(
        measurement_plugin_client_module: ModuleType,
    ) -> None:
        test_measurement_client_type = getattr(measurement_plugin_client_module, "TestMeasurement")
        measurement_plugin_client = test_measurement_client_type()

        with pytest.raises(grpc.RpcError) as exc_info:
            with concurrent.futures.ThreadPoolExecutor() as executor:
                measure = executor.submit(measurement_plugin_client.measure)
                measurement_plugin_client.cancel()
>               measure.result()
E               Failed: DID NOT RAISE <class 'grpc.RpcError'>

tests\acceptance\test_streaming_measurement_client.py:81: Failed
-------------------------------------------------- Captured log call --------------------------------------------------
DEBUG    ni_measurement_plugin_sdk_service.discovery._support:_support.py:37 Discovery service key file path: C:\ProgramData\National Instruments\MeasurementLink\Discovery\v1\DiscoveryService.json
DEBUG    ni_measurement_plugin_sdk_service.grpc.loggers:loggers.py:287 gRPC client call starting: /ni.measurementlink.discovery.v1.DiscoveryService/ResolveService
DEBUG    ni_measurement_plugin_sdk_service.grpc.loggers:loggers.py:291 gRPC client call complete: /ni.measurementlink.discovery.v1.DiscoveryService/ResolveService
DEBUG    ni_measurement_plugin_sdk_service.grpc.loggers:loggers.py:287 gRPC client call starting: /ni.measurementlink.measurement.v2.MeasurementService/Measure
DEBUG    ni_measurement_plugin_sdk_service.grpc.loggers:loggers.py:314 gRPC server call starting: /ni.measurementlink.measurement.v2.MeasurementService/Measure
DEBUG    ni_measurement_plugin_sdk_service.grpc.loggers:loggers.py:337 gRPC server call streaming response: /ni.measurementlink.measurement.v2.MeasurementService/Measure
DEBUG    ni_measurement_plugin_sdk_service.grpc.loggers:loggers.py:299 gRPC client call streaming response: /ni.measurementlink.measurement.v2.MeasurementService/Measure
DEBUG    ni_measurement_plugin_sdk_service.grpc.loggers:loggers.py:337 gRPC server call streaming response: /ni.measurementlink.measurement.v2.MeasurementService/Measure
DEBUG    ni_measurement_plugin_sdk_service.grpc.loggers:loggers.py:299 gRPC client call streaming response: /ni.measurementlink.measurement.v2.MeasurementService/Measure
DEBUG    ni_measurement_plugin_sdk_service.grpc.loggers:loggers.py:337 gRPC server call streaming response: /ni.measurementlink.measurement.v2.MeasurementService/Measure
DEBUG    ni_measurement_plugin_sdk_service.grpc.loggers:loggers.py:299 gRPC client call streaming response: /ni.measurementlink.measurement.v2.MeasurementService/Measure
DEBUG    ni_measurement_plugin_sdk_service.grpc.loggers:loggers.py:337 gRPC server call streaming response: /ni.measurementlink.measurement.v2.MeasurementService/Measure
DEBUG    ni_measurement_plugin_sdk_service.grpc.loggers:loggers.py:299 gRPC client call streaming response: /ni.measurementlink.measurement.v2.MeasurementService/Measure
DEBUG    ni_measurement_plugin_sdk_service.grpc.loggers:loggers.py:337 gRPC server call streaming response: /ni.measurementlink.measurement.v2.MeasurementService/Measure
DEBUG    ni_measurement_plugin_sdk_service.grpc.loggers:loggers.py:299 gRPC client call streaming response: /ni.measurementlink.measurement.v2.MeasurementService/Measure
DEBUG    ni_measurement_plugin_sdk_service.grpc.loggers:loggers.py:337 gRPC server call streaming response: /ni.measurementlink.measurement.v2.MeasurementService/Measure
DEBUG    ni_measurement_plugin_sdk_service.grpc.loggers:loggers.py:299 gRPC client call streaming response: /ni.measurementlink.measurement.v2.MeasurementService/Measure
DEBUG    ni_measurement_plugin_sdk_service.grpc.loggers:loggers.py:337 gRPC server call streaming response: /ni.measurementlink.measurement.v2.MeasurementService/Measure
DEBUG    ni_measurement_plugin_sdk_service.grpc.loggers:loggers.py:299 gRPC client call streaming response: /ni.measurementlink.measurement.v2.MeasurementService/Measure
DEBUG    ni_measurement_plugin_sdk_service.grpc.loggers:loggers.py:337 gRPC server call streaming response: /ni.measurementlink.measurement.v2.MeasurementService/Measure
DEBUG    ni_measurement_plugin_sdk_service.grpc.loggers:loggers.py:299 gRPC client call streaming response: /ni.measurementlink.measurement.v2.MeasurementService/Measure
DEBUG    ni_measurement_plugin_sdk_service.grpc.loggers:loggers.py:337 gRPC server call streaming response: /ni.measurementlink.measurement.v2.MeasurementService/Measure
DEBUG    ni_measurement_plugin_sdk_service.grpc.loggers:loggers.py:299 gRPC client call streaming response: /ni.measurementlink.measurement.v2.MeasurementService/Measure
DEBUG    ni_measurement_plugin_sdk_service.grpc.loggers:loggers.py:337 gRPC server call streaming response: /ni.measurementlink.measurement.v2.MeasurementService/Measure
DEBUG    ni_measurement_plugin_sdk_service.grpc.loggers:loggers.py:299 gRPC client call streaming response: /ni.measurementlink.measurement.v2.MeasurementService/Measure
INFO     ni_measurement_plugin_sdk_service.grpc.loggers:loggers.py:323 gRPC server call /ni.measurementlink.measurement.v2.MeasurementService/Measure responded OK in 10055.6792 ms
DEBUG    ni_measurement_plugin_sdk_service.grpc.loggers:loggers.py:329 gRPC server call complete: /ni.measurementlink.measurement.v2.MeasurementService/Measure
DEBUG    ni_measurement_plugin_sdk_service.grpc.loggers:loggers.py:291 gRPC client call complete: /ni.measurementlink.measurement.v2.MeasurementService/Measure
------------------------------------------------ Captured log teardown ------------------------------------------------
DEBUG    ni_measurement_plugin_sdk_service.grpc.loggers:loggers.py:287 gRPC client call starting: /ni.measurementlink.discovery.v1.DiscoveryService/UnregisterService
DEBUG    ni_measurement_plugin_sdk_service.grpc.loggers:loggers.py:291 gRPC client call complete: /ni.measurementlink.discovery.v1.DiscoveryService/UnregisterService
INFO     ni_measurement_plugin_sdk_service.discovery._client:_client.py:196 Successfully unregistered with discovery service.
INFO     ni_measurement_plugin_sdk_service._internal.service_manager:service_manager.py:165 Measurement service closed.
=============================================== short test summary info ===============================================
FAILED tests/acceptance/test_streaming_measurement_client.py::test___non_streaming_measurement_execution___cancel___cancels_measurement[79-10000] - Failed: DID NOT RAISE <class 'grpc.RpcError'>
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
=================================== 1 failed, 78 passed, 50000 deselected in 16.15s ===================================
```

--- a/packages/generator/tests/acceptance/test_streaming_measurement_client.py
+++ b/packages/generator/tests/acceptance/test_streaming_measurement_client.py
@@ -73,13 +73,17 @@ def test___non_streaming_measurement_execution___cancel___cancels_measurement(
 ) -> None:
     test_measurement_client_type = getattr(measurement_plugin_client_module, "TestMeasurement")
     measurement_plugin_client = test_measurement_client_type()
+    streaming_data_measurement.measurement_started_event.clear()
+    is_canceled = False
 
     with pytest.raises(grpc.RpcError) as exc_info:
         with concurrent.futures.ThreadPoolExecutor() as executor:
             measure = executor.submit(measurement_plugin_client.measure)
-            measurement_plugin_client.cancel()
+            streaming_data_measurement.measurement_started_event.wait()
+            is_canceled = measurement_plugin_client.cancel()
             measure.result()
 
+    assert is_canceled
     assert exc_info.value.code() == grpc.StatusCode.CANCELLED
 
 
@@ -88,13 +92,17 @@ def test___streaming_measurement_execution___cancel___cancels_measurement(
 ) -> None:
     test_measurement_client_type = getattr(measurement_plugin_client_module, "TestMeasurement")
     measurement_plugin_client = test_measurement_client_type()
+    streaming_data_measurement.measurement_started_event.clear()
+    is_canceled = False
 
     with pytest.raises(grpc.RpcError) as exc_info:
         with concurrent.futures.ThreadPoolExecutor() as executor:
             measure = executor.submit(lambda: list(measurement_plugin_client.stream_measure()))
-            measurement_plugin_client.cancel()
+            streaming_data_measurement.measurement_started_event.wait()
+            is_canceled = measurement_plugin_client.cancel()
             measure.result()
 
+    assert is_canceled
     assert exc_info.value.code() == grpc.StatusCode.CANCELLED
 
 

--- a/packages/generator/tests/utilities/measurements/streaming_data_measurement/__init__.py
+++ b/packages/generator/tests/utilities/measurements/streaming_data_measurement/__init__.py
@@ -21,6 +21,8 @@ measurement_service = nims.MeasurementService(
 
 Outputs = tuple[str, int, list[int]]
 
+measurement_started_event = threading.Event()
+
 
 @measurement_service.register_measurement
 @measurement_service.configuration("name", nims.DataType.String, "<Name>")
@@ -41,6 +43,8 @@ def measure(
     error_on_index: int,
 ) -> Generator[Outputs]:
     """Returns the number of responses requested at the requested interval."""
+    measurement_started_event.set()
+
     cancellation_event = threading.Event()
     measurement_service.context.add_cancel_callback(cancellation_event.set)
 

